### PR TITLE
[Single] Added unittests

### PR DESCRIPTION
**Pull request by kparichay**
Added more unittests for single-shot API testing
1. Multiple runs in parallel with varying timeouts
2. Changing timeout/closing the handle while calling the API in a different thread
3. Closing the handle multiple times
4. Check valid result after increasing the timeout, when earlier it was timing out

**Self evaluation:**
1. Build test: [ ]Passed [ ]Failed [*]Skipped
2. Run test: [ ]Passed [ ]Failed [* ]Skipped

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

File: api/capi/src/nnstreamer-capi-single-new.c [hunk 1]
```diff
@@ -44,18 +46,27 @@
   (ts).tv_nsec = ((msec) % 1000) * 1000000; \
 } while (0)
 
+/** verify the magic number for ml_single obj */
+#define ML_SINGLE_MAGIC_CHECK(arg) do { \
+  if (arg->magic != ML_SINGLE_MAGIC) { \
+    ml_loge ("The given param, single is invalid."); \
+    return ML_ERROR_INVALID_PARAMETER; \
+  } \
+} while (0)
```
Comment (myungjoo) [inline api/capi/src/nnstreamer-capi-single-new.c]: It'd be better to define in a common header for both C files.

Comment (kparichay) [reply api/capi/src/nnstreamer-capi-single-new.c]: A common header would introduce big changes, irrelevant to this PR. How about a new PR for that?

Comment (myungjoo) [reply api/capi/src/nnstreamer-capi-single-new.c]: Ok.

Please note that you may place an internal header at /api/capi/src/
or
Let nnstreamer-capi-single.c include tensor_filter_single.h, which minimizes changes.

File: api/capi/src/nnstreamer-capi-single.c [hunk 2]
```diff
@@ -431,6 +443,9 @@ ml_single_close (ml_single_h single)
   }
 
   single_h = (ml_single *) single;
+  ML_SINGLE_MAGIC_CHECK (single_h);
```
Comment (jaeyun-jung) [inline api/capi/src/nnstreamer-capi-single.c]: Discussion about validating the handle (maybe TODO list)
How about checking magic inside the mutex lock?
IMO, single_h->is_valid is unnecessary with single-magic.

File: api/capi/src/nnstreamer-capi-single-new.c (hunk 1 above)
Comment (kparichay) [reply api/capi/src/nnstreamer-capi-single-new.c]: I will add after #1672 is merged

File: api/capi/src/nnstreamer-capi-single.c (hunk 2 above)
Comment (kparichay) [reply api/capi/src/nnstreamer-capi-single.c]: `is_valid` has been removed
I think checking inside lock will assume that the lock is valid. However, if the object is invalid, then the lock can be invalid as well, which can cause the program to crash when trying to acquire the lock. Please correct me if I am wrong.

File: api/capi/src/nnstreamer-capi-single.c [hunk 3]
```diff
@@ -401,7 +411,6 @@ ml_single_open (ml_single_h * single, const char *model,
   }
 
   g_mutex_lock (&single_h->lock);
```
Comment (jaeyun-jung) [inline api/capi/src/nnstreamer-capi-single.c]: plz remove these two lines also (nothing to do here)

Comment (kparichay) [reply api/capi/src/nnstreamer-capi-single.c]: done.

File: api/capi/src/nnstreamer-capi-single.c [hunk 4]
```diff
@@ -559,7 +561,11 @@ ml_single_invoke (ml_single_h single,
   if (!sample) {
     ml_loge ("Failed to get the result from sink element.");
     single_h->clear_previous_buffer = TRUE;
+#ifdef ML_SINGLE_SUPPORT_TIMEOUT
```
Comment (jaeyun-jung) [inline api/capi/src/nnstreamer-capi-single.c]: inside the feature (appsink without timeout feature does not need to clear old buffer)

```
#ifdef ML_SINGLE_SUPPORT_TIMEOUT
    single_h->clear_previous_buffer = TRUE;
```

Comment (kparichay) [reply api/capi/src/nnstreamer-capi-single.c]: Right, updated.

**Pull request merged**